api/sshkeys/service.go

```diff
@@ -49,10 +50,14 @@ func (s *service) CreatePublicKey(ctx context.Context, key *models.PublicKey) er
 
 	key.Fingerprint = ssh.FingerprintLegacyMD5(pubKey)
 
-	err = s.store.CreatePublicKey(ctx, key)
-	if err == store.ErrDuplicateFingerprint {
-		return ErrDuplicateFingerprint
+	returnedKey, err := s.store.GetPublicKey(ctx, key.Fingerprint, apicontext.TenantFromContext(ctx).ID)
+	if err == store.ErrRecordNotFound && returnedKey == nil {
```

Comment:
Just check the err here as if this returns, we ought to break out.

Reply:
we can't just return the error here, because the `recordNotFound` is a expected error, means there are no registered public keys with the same fingerprint and tenant_id.